artiq/firmware/libboard_artiq/si5324.rs

```diff
@@ -179,17 +179,17 @@ fn init() -> Result<()> {
 
     #[cfg(soc_platform = "kasli")]
     {
-        i2c::pca9548_select(BUSNO, 0x70, 0)?;
-        i2c::pca9548_select(BUSNO, 0x71, 1 << 3)?;
+        i2c::switch_select(BUSNO, 0x70 << 1, 0)?;
```

Comment:
I would keep the 7-bit address here which is what the rest of the firmware does IIRC. Modify the kernel side as appropriate.